p2p/server.go

```diff
@@ -200,7 +203,7 @@ func makeBasicHost(ctx context.Context, priv crypto.PrivKey, pub crypto.PubKey,
 		libp2p.AddrsFactory(addressFactory),
 	}
 
-	bhost, err := libp2p.New(ctx, opts...)
+	bhost, err := libp2p.New(opts...)
```

Comment:
Since `libp2p.New` doesnt take a context anymore, I am wondering if they replace some of the usages of ctx by new `opts` that we may have to set. As far as I know we used to pass a cancel context but that might not be needed there anymore? How would the cancel cascading works now?